lib/merkleproofs.js

```diff
@@ -1,7 +1,16 @@
-const merkleproofs = {
+const DashUtil = require('@dashevo/dash-util');
 
-  validateTxProofs: (merkleBlock, transactions) => merkleBlock.validMerkleTree()
-    && transactions.filter(t => merkleBlock.hasTransaction(t)).length === transactions.length,
+const merkleproofs = {
+  validateTxProofs: (merkleBlock, transactions) => {
+    let txToFilter = [];
+    if (typeof transactions[0] === 'string') {
+      txToFilter = transactions.slice().map(tx => DashUtil.toHash(tx).toString('hex'));
```

Comment:
It seems you need to make `transactions.slice()` anyways. What if just put it to variable initialization?
Like:
```js
const txToFilter = transactions.slice();
```

Reply:
done